partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/RexConverter.kt

```diff
@@ -860,13 +865,20 @@ internal object RexConverter {
                 DataType.BIT -> call(FunctionUtils.OP_IS_BIT, arg0) // TODO define in parser
                 DataType.BIT_VARYING -> call(FunctionUtils.OP_IS_BIT_VARYING, arg0) // TODO define in parser
                 // <numeric types> - <exact numeric types>
-                DataType.NUMERIC -> call(FunctionUtils.OP_IS_NUMERIC, targetType.precision.toRex(), targetType.scale.toRex(), arg0)
-                DataType.DEC, DataType.DECIMAL -> call(
-                    FunctionUtils.OP_IS_DECIMAL,
-                    targetType.precision.toRex(),
-                    targetType.scale.toRex(),
-                    arg0
-                )
+                DataType.NUMERIC, DataType.DEC, DataType.DECIMAL -> {
+                    when {
+                        targetType.precision == null && targetType.scale == null -> call(
```

Comment:
if precision is non-null, then scale is guaranteed to be non-null?

Reply:
following our conversation, I checked this again and seems like I was wrong. the null scale conversion only applied for `CAST`s and not for `IS TYPE`. in the latest commit, I added the conversion for `null` scale to be a scale of `0` for `IS TYPE` -- [`65fef33` (#1749)](https://github.com/partiql/partiql-lang-kotlin/pull/1749/commits/65fef335b2126f341696eab808e331b5a1c36aa7)